src/main.rs

```diff
@@ -50,6 +49,26 @@ macro_rules! plural {
     };
 }
 
+fn is_rept(node: &Node) -> bool {
+    matches!(node.type_data(), NodeType::Rept(..))
+}
+
+fn rept_parent_non_rept<'a>(object: &'a Object, node: &'a Node) -> Option<&'a Node> {
+    if is_rept(node) {
+        node.parent().and_then(|(parent_id, _)| {
+            object.node(parent_id).and_then(|parent| {
+                if is_rept(parent) {
+                    rept_parent_non_rept(object, parent)
```

Comment:
Recursion is not a great idea, since tail calls are not guaranteed; consider making `node` `mut`, and using a loop?

Reply:
Sure.